### PR TITLE
submitted th_TH providers

### PR DESCRIPTION
**Pull request by ppeemk**
fixed ref name

### CODE REVIEW

File: faker/providers/person/th_TH/__init__.py [hunk 1]
```diff
@@ -62,29 +68,87 @@ class Provider(PersonProvider):
...
-        "ผจงรักษ์", "ภัคจิรา", "ปัทธมพร", "สุวิจักษ์", "นพเก้า", "รัชชานนท์", "อธิศ",
-        "คุณาพร", "อริสรา", "ทัศวรรณ", "ชัยวัฒน์", "ภัณฑิรา", "ศุภสุตา", "พัชรพร",
-        "ภูเบศ",
-        "พัทธมน", "ชัญญา", "ปัณณวิชย์", "ชวิน", "ภาดา", "ชนิกานต์", "ณัฐวดี", "ชาลิสา",
-        "ประเมศฐ์", "ศุภสิทธิ์", "สิทธิวัต", "ธีธีช", "ญาดา", "เพ็ญพิชชา", "อนนต์",
-        "ชายฟ้า",
-        "ปวัน", "อัญญาร", "ญานิศา", "ภาคย์", "ชญานินท์", "ชยาภัทร", "จิตริน", "วัศยา",
-        "ปิติภัทร", "นิชากานต์", "ปริน", "ธนัชชา",)
+    # female first name
+    # http: // www.mamaexpert.com / posts / content - 1166
```
Comment (fcurella) [inline faker/providers/person/th_TH/__init__.py]: Use actual URL, without the spaces. Also, add a brief description of the source.

Comment (fcurella) [inline faker/providers/person/th_TH/__init__.py]: I'd rather use more official sources, like a Vital Statistics Agency or a census. A wikipedia page in any language would work to.

File: faker/providers/person/th_TH/__init__.py [hunk 2]
```diff
@@ -62,29 +68,87 @@ class Provider(PersonProvider):
...
+        "ปลื้ม", "ปุณ", "ปุณณ์", "ปุญญ์", "แทนคุณ", "ธัน", "ธันวา", "พอร์ช", "พัตเตอร์",
+        "พอร์ตเตอร์", "พอตเตอร์", "ฟลุ๊ก", "ฟลุ๊ค", "เฟสบุ๊ค", "ภาคิน", "ภูมิ", "ภูผา",
+        "ภาคภูมิ", "ภู", "ภูมิใจ", "มาวิน", "มีคุณ", "ออโต้", "ออดี้", "อิ๊คิว", "อิคคิว",
+        "อิคคิวซัง", "อิกคิว", "อะตอม", "ไอซ์", "อั่งเปา", "ออกัส", "อชิ", "อะชิ",
+        "เอิร์ธ",)
+
+    first_names = first_names_female + first_names_male
+
+    # last name
+    # http://www.reurnthai.com/index.php?topic=5273.45
```
Comment (fcurella) [inline faker/providers/person/th_TH/__init__.py]: I'm not sure what this url is. Seems to be a forum thread about literature. I think a comment explaining why you're picking names from the there would be helpful, for people not familiar with Thai culture.

File: faker/providers/person/th_TH/__init__.py [hunk 3]
```diff
@@ -62,29 +68,87 @@ class Provider(PersonProvider):
...
+        "นาถะเดชะ",
+        "นพตระกูล", "นิยมเซียม", "หนักแน่น", "นามขำ", "นานายน", "นับเนื่องนอ", "แนวพญา",
+        "นาถะพินธุ", "นิลสลัว", "นะวะมันดร", "นิลวรรณ", "เนตร์มณี", "นุตตาร",
+        "นิยมสำหรวจ",
+        "นกทอง", "นุชแนวนุ่ม", "นาคสุทิน", "นพคเชนทร์", "นาคะนคร", "นิละทัต", "นาควงษ์",
+        "นากกนก", "บุนยาภิสนท์", "บุญญาภิรมย์", "บุณยะภาชน์", "บินดี", "บุนยะศัพท์",
+        "บุนยะตีรณะ", "บุญญาไลย์", "บุญบำรุง",)
+
+    # thai suffix
+    # https://th.wikipedia.org/wiki/%E0%B8%99%E0%B8%B2%E0%B8%A1%E0%B8%AA%E0%B8%81%E0%B8%B8%E0%B8%A5%E0%B8%9E%E0%B8%A3%E0%B8%B0%E0%B8%A3%E0%B8%B2%E0%B8%8A%E0%B8%97%E0%B8%B2%E0%B8%99
```
Comment (fcurella) [inline faker/providers/person/th_TH/__init__.py]: By translating the page with google, this appears to be a list of last names, not suffixes.

Comment (ppeemk) [inline faker/providers/person/th_TH/__init__.py]: Actually this is like the last name that has been given by the king , but nowadays people will use it as a prefix for example Jiratit Tinnakorn na Ayuddhaya ; Jiratit <- first name , Tinnakorn <- last name , na Ayuddhaya <- works like prefix

File: faker/providers/person/th_TH/__init__.py (hunk 1 above)
Comment (ppeemk) [inline faker/providers/person/th_TH/__init__.py]: will find more official sources :)

**Comment by ppeemk** [comment]
Anything i should have fixed?

**Comment by fcurella** [comment]
@ppeemk I think it's good enough. Thank you so much!

**Pull request merged**